client/signup/steps/site-type/index.jsx

```diff
@@ -26,45 +27,53 @@ import FormRadio from 'components/forms/form-radio';
 class SiteType extends Component {
 	constructor( props ) {
 		super( props );
-		const siteTypeVal = isValidLandingPageSiteType( props.siteType ) ? props.siteType : '';
 		this.state = {
-			siteType: siteTypeVal,
+			siteType: props.siteType,
 		};
 	}
 
+	componentDidMount() {
+		SignupActions.saveSignupStep( {
+			stepName: this.props.stepName,
+		} );
+	}
+
 	handleRadioChange = event => this.setState( { siteType: event.currentTarget.value } );
 
 	handleSubmit = event => {
 		event.preventDefault();
-		// Default siteType is 'blogger'
-		const siteTypeInputVal = this.state.siteType || allSiteTypes[ 0 ].type;
+		// Default siteType is 'blog'
+		const siteTypeInputVal = this.state.siteType || get( allSiteTypes, 'blog.value', '' );
 		const themeRepo = getThemeForSiteType( siteTypeInputVal );
 
 		this.props.submitStep( siteTypeInputVal, themeRepo );
 	};
 
-	renderContent() {
-		const { translate } = this.props;
-		const radioOptions = allSiteTypes.map( elem => (
-			<FormLabel className="site-type__option" key={ elem.type }>
+	renderRadioOptions() {
+		return Object.values( allSiteTypes ).map( elem => (
```

Comment:
If we go with the `array/object` structure proposed above, this could just be `allSiteTypes.map( elem => ( //...`